core-bundle/src/EventListener/DataContainer/Undo/FromTableOptionsListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer\Undo;
+
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @Callback(table="tl_undo", target="fields.options.fromTable")
+ *
+ * @internal
+ */
+class FromTableOptionsListener
+{
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function __invoke(): array
+    {
+        $tables = $this->connection->executeQuery('SELECT DISTINCT '.$this->connection->quoteIdentifier('fromTable').' FROM tl_undo');
+
+        if (0 === $tables->rowCount()) {
+            return [];
+        }
+
+        $options = [];
+
+        foreach ($tables->fetchFirstColumn() as $table) {
+            $options[] = $table;
+        }
+
+        return $options;
+    }
```

Comment:
Could this be shortened? :upside_down_face: 
```suggestion
        return $this->connection
            ->executeQuery('SELECT DISTINCT '.$this->connection->quoteIdentifier('fromTable').' FROM tl_undo')
            ->fetchFirstColumn()
        ;
```

Reply:
While trying to answer your question, I noticed that  the listener is actually never used. 🤔 Since undo records cannot be edited, I wonder what the listener is used for?

Reply:
I just realised this is obsolete now. It is from when we had translations for tables and these were spread out across multiple translation files. I used the `options_callback` to gather the translations and use them for the filter menu. 

Removed in fe4f685c9